test/unit/MapIterableCapableTraitTest.php

```diff
@@ -0,0 +1,526 @@
+
+        return $mock;
+    }
+
+    /**
+     * Creates a new Invalid Argument exception.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $message The exception message.
+     *
+     * @return MockObject|InvalidArgumentException The new exception.
+     */
+    public function createInvalidArgumentException($message = '')
+    {
+        $mock = $this->getMockBuilder('InvalidArgumentException')
+            ->setConstructorArgs([$message])
+            ->getMock();
+
+        return $mock;
+    }
+
+    /**
+     * Creates a new Invocation exception.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $message The exception message.
+     *
+     * @return MockObject|InvocationExceptionInterface The new exception.
+     */
+    public function createInvocationException($message = '')
+    {
+        $mock = $this->mockClassAndInterfaces('Exception', ['Dhii\Invocation\Exception\InvocationExceptionInterface'])
+            ->setConstructorArgs([$message])
+            ->getMock();
+
+        return $mock;
+    }
+
+    /**
+     * Creates a new `Traversable` mock.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $elements The elements for the traversable.
+     *
+     * @return MockObject|ArrayIterator
+     */
+    public function mockTraversable($elements = [])
+    {
+        $mock = $this->getMockBuilder('ArrayIterator')
+            ->setConstructorArgs([$elements])
+            ->enableProxyingToOriginalMethods()
+            ->getMock();
+
+        return $mock;
+    }
+
+    /**
+     * Tests whether a valid instance of the test subject can be created.
+     *
+     * @since [*next-version*]
+     */
+    public function testCanBeCreated()
+    {
+        $subject = $this->createInstance();
+
+        $this->assertInternalType(
+            'object',
+            $subject,
+            'A valid instance of the test subject could not be created.'
+        );
+    }
+
+    /**
+     * Tests that `_mapIterable()` works as expected when given an array.
+     *
+     * @since [*next-version*]
+     */
+    public function testMapIterableArray()
+    {
+        $key1 = uniqid('key');
+        $val1 = uniqid('val');
+        $key2 = uniqid('key');
+        $val2 = uniqid('val');
+        $key3 = uniqid('key');
+        $val3 = uniqid('val');
+        $data = [
+            $key1 => $val1,
+            $key2 => $val2,
+            $key3 => $val3,
+        ];
+        $start = 0;
+        $end = 0;
+        $iterable = $data;
+        $subject = $this->createInstance();
+        $_subject = $this->reflect($subject);
+
+        $cb = function ($value, $key, $iterator) {
+            return implode('-', [$key, $value]);
+        };
+        $expected = [
+            $key1 => implode('-', [$key1, $val1]),
+            $key2 => implode('-', [$key2, $val2]),
+            $key3 => implode('-', [$key3, $val3]),
+        ];
+
+        $subject->expects($this->exactly(1))
```

Comment:
So lately I've developed a bit of a habit. Maybe it might interest you.

Unless it's something explicitly obvious, like expecting `_invokeCallable` exactly **3** times, or unless the method being expected has some permanent consequence (like executing a DB query), I try to keep the expectation as loose as possible, using things like `$this->atLeast($c)` or `$this->atLeastOnce()`.

The reason being that it might not be important how many times the method is invoked, but instead it's important that it _is_ invoked and that it returns some mock result.

Reply:
In my experience, in most cases it is important exactly how many times a method is executed. I think that this case is no exception.